pkg/utils/provider_meta.go

```diff
@@ -110,3 +112,13 @@ func ExtractId(oldID string) string {
 	}
 	return parts[1]
 }
+
+// Function to get the region from the region + ID string
+func ExtractRegion(oldID string) string {
+	parts := strings.Split(oldID, ":")
+	if len(parts) < 2 {
+		// Return an empty string if the ID doesn't have a region
+		return ""
```

Comment:
Unfortunately, in Go, you cannot return `nil` for a value that is expected to be a `string`. It would error out with:

```
cannot use nil as string value in return statement compiler
```